tests/e2e/bulker.test.ts

```diff
@@ -0,0 +1,1026 @@
+    );
+
+    const userWethBalance = await weth.balanceOf(address);
+    expect(userWethBalance).to.be.equal(
+      initialWethBalance,
+      `expect weth wallet balance is ${initialWethBalance}, received ${initialWethBalance}`
+    );
+
+    const bulkerWethAllowance = await wethAllowance();
+    expect(bulkerWethAllowance).to.equal(
+      constants.MaxUint256,
+      `expected impersonated user weth allowance is ${constants.MaxUint256}, received ${bulkerWethAllowance}`
+    );
+
+    const userDaiBalance = await dai.balanceOf(address);
+    expect(userDaiBalance).to.be.equal(
+      initialDaiBalance,
+      `expect user dai balance is ${initialDaiBalance}, received ${userDaiBalance}`
+    );
+
+    const bulkerDaiAllowance = await daiAllowance();
+    expect(bulkerDaiAllowance).to.equal(
+      constants.MaxUint256,
+      `expected impersonated user dai allowance is ${constants.MaxUint256}, received ${bulkerDaiAllowance}`
+    );
+
+    const bulkerStEthAllowance = await stethAllowance();
+    expect(bulkerStEthAllowance).to.equal(
+      constants.MaxUint256,
+      `expected impersonated user steth allowance is ${constants.MaxUint256}, received ${bulkerStEthAllowance}`
+    );
+
+    const bulkerWstEthAllowance = await wstethAllowance();
+    expect(bulkerWstEthAllowance).to.equal(
+      constants.MaxUint256,
+      `expected impersonated user wsteth allowance is ${constants.MaxUint256}, received ${bulkerWstEthAllowance}`
+    );
+
+    const stEthBalance = await steth.balanceOf(address);
+    expect(
+      approxEqual(stEthBalance, initialStEthBalance),
+      `approximate expected user steth balance is ${initialStEthBalance}, received ${stEthBalance}`
+    ).to.be.true;
+  });
+
+  const toAllow = [CONTRACT_ADDRESSES.bulker, CONTRACT_ADDRESSES.permit2];
+  toAllow.forEach((contractAddress) => {
+    const approvalType =
+      contractAddress === CONTRACT_ADDRESSES.bulker ? "Bulker" : "Permit2";
+
+    describe(`Supply transaction with ${approvalType} approval`, () => {
+      it("Should supply DAI as collateral", async () => {
+        await approveBulkerOrPermit2(contractAddress);
+        const maxDaiCapacity = bulker.getUserMaxCapacity(
+          Underlying.dai,
+          TransactionType.supplyCollateral
+        )!;
+
+        await bulker.addOperation({
+          type: TransactionType.supplyCollateral,
+          amount: maxDaiCapacity.amount,
+          underlyingAddress: Underlying.dai,
+        });
+
+        for (const signature of bulker.signatures$.getValue())
+          await bulker.sign(signature);
+        await bulker.executeBatch();
+
+        expect(maxDaiCapacity.limiter).to.equal(
+          MaxCapacityLimiter.walletBalance
+        );
+
+        const userDaiBalance = await dai.balanceOf(morphoUser.address);
+        expect(userDaiBalance).to.be.equal(
+          constants.Zero,
+          `expected user dai balance is ${constants.Zero}, received ${userDaiBalance}`
```

Comment:
```suggestion
          `expected user dai balance is ${constants.Zero.toString()}, received ${userDaiBalance.toString()}`
```

Reply:
It will call `.toString` on the BigNumbers if this is omitted.